Makefile

```diff
@@ -0,0 +1,20 @@
+golangci-lint-version = v1.56.1
+ifeq ($(GOOS),windows)
+  WORKSPACE := ${shell go env GOPATH}
+else
+  WORKSPACE := ${shell go env GOPATH | rev | cut -d: -f1 | rev}
+endif
```

Comment:
It's not clear to me what this conditional is doing, since `make ci-lint` always runs three times for each GOOS. But GOPATH will be the same each time, won't it?

In any case, I wonder if it would be safer to just install to a `bin` folder under the git repo instead of trying to install globally? Then we're not overwriting whichever version of golangci-lint the user happens to have installed.

Reply:
1. Yeah, it might be a little too preventive to call the linter for each GOOS now as currently there is no system related things there, but better to be safe than sorry. And yes, the linter can have different output when run with different os types.

2. Good idea, done

Reply:
Ah, I actually misread this originally. For some reason, I thought the three lines at the bottom were recursively calling this makefile with a different GOOS each time. But that's not what they're doing at all. Makes way more sense to me now. 😄

Now, as far as I can tell, there's no way to run a Makefile without first having to `cd` into the Makefile's directory. Even the `-C` option works by changing the current directory as the first step. So that means we can simplify this further, using `.` (or just specifying relative paths like `bin` and `bin/golangci-lint`) instead of needing this WORKSPACE variable at all. Does that make sense?

That would ensure everything behaves the same on Windows, too.

Reply:
It think we have to do it the way it is now. I've tried make the linter run on some specific directory but it fails. Running the linter inside the directory runs just fine (it seems like a `golangci-lint` to me but I might be misinterpreting something). Can we leave it like this for now?

Reply:
Just noting this was resolved in 5f0d669

It might also be useful to add a basic .gitignore now too. Something like:

```
*.swp
/bin
```

Reply:
added `.gitignore` with:
```
*.swp
/bin
```